### PR TITLE
fix: don't cancel ticket in past

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/6497

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/account/danger-zone.js [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 export default class DangerZone extends Component {
   @computed('data.events', 'data.orders')
   get isUserDeletable() {
-    if (this.data.events.length || this.data.orders.length) {
+    let validOrder = 0;
+    this.data.orders.map(x => {
+      if (x.get('status') !== 'cancelled' && x.event.get('endsAt') > moment()) {
+        validOrder = validOrder + 1;
+      }
+    });
```
Comment (iamareebjamal) [inline app/components/account/danger-zone.js:14]: Don't calculate here, make an endpoint in server to reuse the logic and use that

File: app/templates/my-tickets/past.hbs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <div class="row">
   <div class="sixteen wide column">
     {{#each this.model as |order|}}
-      <OrderCard @order={{order}}>
+      <OrderCard @order={{order}} @noCanelTicket={{true}}>
```
Comment (iamareebjamal) [inline app/templates/my-tickets/past.hbs:4]: What is canelTicket?

File: app/components/account/danger-zone.js [hunk 3]
```diff
@@ -1,12 +1,18 @@
 import classic from 'ember-classic-decorator';
 import { action, computed } from '@ember/object';
 import Component from '@ember/component';
-
+import moment from 'moment';
```
Comment (iamareebjamal) [inline app/components/account/danger-zone.js:4]: newline missing

File: app/routes/account/danger-zone.js [hunk 4]
```diff
@@ -12,11 +13,23 @@ export default class DangerZoneRoute extends Route.extend(AuthenticatedRouteMixi
...
+    },
+    {
+      name : 'event',
+      op   : 'has',
+      val  : {
+        name : 'ends-at',
+        op   : 'ge',
+        val  : moment()
+      }
+    }]
```
Comment (iamareebjamal) [inline app/routes/account/danger-zone.js:32]: indentation

**Comment by maze-runnar** [comment]
this PR also consist of not showing cancelOrder button in pastEvent section.

**Comment by iamareebjamal** [comment]
OK, don't set Fixes to PRs, only issues

**Pull request merged**